kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -137,6 +137,9 @@ trait ProcBuilder {
 
     val envs = pb.environment()
     envs.putAll(env.asJava)
+    if (shortName.equals("flink")) {
```

Comment:
we can move this to flink-proc-builder？

Reply:
There is childProcEnv in the history, but it is removed from this PR [https://github.com/apache/incubator-kyuubi/pull/2572](https://github.com/apache/incubator-kyuubi/pull/2572). Can I add it again?

Reply:
maybe we can put it into `pb.environment()` directly

Reply:
Would it be better to move HADOOP_PROXY_USER to conf of FlinkProcessBuilder?